### DISCUSSION TITLE
Ocsp response containing multiple certificates

### DISCUSSION THREAD

**Discussion by overocean** [randombit/botan#4497] [Q&A] [answered]
Hi guys, I found that when we construct OCSP Response, we call `decode_optional_list()` to construct the certificates. Inside the body of the while loop in this function, `certbits` will be the sequence of certificates in the response data, and the X509_Certificate constructor will construct an object with the first certificate data in `certbits`. Therefore, other certificates are discarded. This seems a bug to me.

```
   BER_Decoder list(obj);

   while(list.more_items()) {
      BER_Object certbits = list.get_next_object();
      X509_Certificate cert(certbits.bits(), certbits.length());
      output.push_back(std::move(cert));
   }

**Comment by reneme** [comment]
Thanks for reporting. I've transformed this into an issue (#4502), to allow for easier tracking. Let's continue the discussion there. I l'll be looking into it in a few days. Thanks for your patience.

**Reply by reneme** [reply, marked as answer]
See #4536 for a fix.
